### PR TITLE
Demonstrate code that only executes in test runtime

### PR DESCRIPTION
**Pull request by chuck-dbos**
Make use of the setConfig testing feature.
Switch to bcrypt communicator

Up-level the packages so these features are available.  This was not without consternation, otel and @types/koa had to be added as [dev] dependencies.

### CODE REVIEW

File: e-commerce/shop-backend/package.json [hunk 1]
```diff
@@ -9,17 +9,20 @@
     "lint-fix": "eslint --fix src"
   },
   "dependencies": {
-    "@dbos-inc/communicator-bcrypt": "^0.8.62",
-    "@dbos-inc/dbos-sdk": "^0.8.62",
+    "@dbos-inc/communicator-bcrypt": "^0.9.25",
+    "@dbos-inc/dbos-sdk": "^0.9.25",
```
Comment (maxdml) [inline e-commerce/shop-backend/package.json:13]: Shouldn't we use preview here too? (To be consistent with the payment backend?)

File: e-commerce/payment-backend/src/operations.ts [hunk 2]
```diff
@@ -170,15 +170,15 @@ export class PlaidPayments {
...
     webhook: string, 
     session_id: string, 
     status: string | undefined, 
     client_reference_id: string | undefined
   ): Promise<void>
   {
-    if (webhook === "http://fakehost/webhook") {
-        return; // In testing - but it would be good to confirm this by checking the environment
+    if (ctxt.getConfig('unittest', false) && webhook === "http://fakehost/webhook") {
+      return; // In testing, matching the bogus testing URL
```
Comment (maxdml) [inline e-commerce/payment-backend/src/operations.ts:181]: That part is interesting. I didn't realize we would have to modify production code for the sake of unit testing. Ideally we would be able to mock the `paymentWebhook` from the testing runtime and selectively have it perform its actual implementation or the mocked behavior, [like jest would let you d](https://jestjs.io/docs/mock-functions#mocking-partials)o.

Comment (chuck-dbos) [reply e-commerce/payment-backend/src/operations.ts:181]: If you look at the PR from yesterday, I did that.  So, we have examples of both.

File: e-commerce/shop-backend/package.json (hunk 1 above)
Comment (chuck-dbos) [reply e-commerce/shop-backend/package.json:13]: It won't be long before the preview is published and we can then do both to latest.

**Pull request merged**